Feat: Default function opacity setting + autoreveal option (#4342)
dafny-lang/dafny: docs/DafnyRef/Types.md

1, or maybe 2 or 3). If its arguments are all literals it is
transparent all the way.

The default transparency of a function can be set with the `--default-function-opacity` commandline flag. By default, the `--default-function-opacity` is transparent.
The transparency of a function is also affected by
whether the function was declared with an `opaque` modifier or [`transparent` attribute](#sec-transparenct),
the ([reveal statement](#sec-reveal-statement)),
and whether it was `reveal`ed in an export set.

Inside the module where the function is declared:
  - If `--default-function-opacity` is set to `transparent` (default), then:
     - if there is no `opaque` modifier, the function is transparent.
     - if there is an `opaque` modifier, then the function is opaque. If the function is mentioned in a `reveal` statement, then

>      it's body is available starting at that `reveal` statement.

Reviewer: ```suggestion
     its body is available starting at that `reveal` statement.
```

>      it's body is available in the callable where the `reveal` statement occurs.

Reviewer 2: ```suggestion
     its body is available starting at that `reveal` statement.
```
